lib/runner.js

```diff
@@ -44,67 +43,82 @@ Runner.prototype.addAfterHook = function (title, cb) {
 	this.tests.after.push(new Test(title, cb));
 };
 
-Runner.prototype.concurrent = function (tests, cb) {
-	each(tests, function (test, i, next) {
-		test.run(function (err, duration) {
-			if (err) {
-				this.stats.failCount++;
-			}
-
-			this.results.push({
-				duration: duration,
-				title: test.title,
-				error: err
-			});
-
-			this.emit('test', err, test.title, duration);
-			next();
-		}.bind(this));
-	}.bind(this), cb);
+Runner.prototype.concurrent = function (tests) {
+	// run all tests
+	tests = tests.map(function (test) {
+		return test.run();
+	});
+
+	return Promise.all(tests, function (test) {
+		return test
+			.catch(function () {
+				return;
+			})
+			.then(function () {
+				if (test.assertError) {
+					this.stats.failCount++;
+				}
+
+				this.results.push({
+					duration: test.duration,
+					title: test.title,
+					error: test.assertError
+				});
+
+				this.emit('test', test.assertError, test.title, test.duration);
+			}.bind(this));
+	}.bind(this));
 };
 
-Runner.prototype.serial = function (tests, cb) {
-	eachSerial(tests, function (test, next) {
-		test.run(function (err, duration) {
-			if (err) {
-				this.stats.failCount++;
-			}
-
-			this.results.push({
-				duration: duration,
-				title: test.title,
-				error: err
-			});
-
-			this.emit('test', err, test.title, duration);
-			next();
-		}.bind(this));
-	}.bind(this), cb);
+Runner.prototype.serial = function (tests) {
+	return Promise.resolve(tests).each(function (test) {
```

Comment:
This inner function can be DRYed up as the `Runner.prototype.concurrent` has the exact same function content.